src/master-playlist-controller.js

```diff
@@ -824,8 +870,10 @@ export class MasterPlaylistController extends videojs.EventTarget {
     nextPlaylist = this.selectPlaylist();
 
     if (nextPlaylist) {
-      videojs.log.warn('Problem encountered with the current ' +
-                       'HLS playlist. Switching to another playlist.');
+      let logMessage = error.message ? 'Problem encountered with the current HLS playlist. ' + error.message + ' Switching to another playlist.' :
```

Comment:
I think this may be cleaner as:

```js
videojs.log.warn('Problem encountered with the current HLS playlist.' +
                 (error.message ? ' ' + error.message : '') +
                 ' Switching to another playlist.');
```